editor/editor_interface.cpp

```diff
@@ -512,6 +513,38 @@ void EditorInterface::popup_quick_open(const Callable &p_callback, const TypedAr
 	quick_open->popup_dialog(base_types, callable_mp(this, &EditorInterface::_quick_open).bind(p_callback));
 }
 
+void EditorInterface::popup_create_dialog(const Callable &p_callback, const StringName &p_base_type, const TypedArray<StringName> &p_custom_type_blacklist, const Dictionary &p_custom_suffix) {
+	if (!create_dialog) {
+		create_dialog = memnew(CreateDialog);
+		get_base_control()->add_child(create_dialog);
+	}
+
+	HashSet<StringName> blacklist;
+	for (int i = 0; i < p_custom_type_blacklist.size(); i++) {
+		blacklist.insert(p_custom_type_blacklist[i]);
+	}
+	create_dialog->set_type_blacklist(blacklist);
+
+	HashMap<StringName, String> suffix_map;
+	for (int j = 0; j < p_custom_suffix.size(); j++) {
+		StringName k = p_custom_suffix.get_key_at_index(j);
+		if (k.is_empty()) {
+			// Skips empty keys and invalid values. The dictionary should be Dictionary[StringName, String],
+			// but due to a bug when using TypedDictionary, the final compile will failed due to being unable to
+			// parse ";" after "typeddictionary::" during the godot-cpp binding.
+			continue;
+		}
+		suffix_map.insert(k, p_custom_suffix[k]);
+	}
+	create_dialog->set_type_suffixes(suffix_map);
+
+	create_dialog->set_base_type(p_base_type.is_empty() ? "Object" : p_base_type);
+	create_dialog->select_base();
+	create_dialog->popup_create(false, true, "", "");
```

Comment:
The first string is argument for currently selected type. Maybe we could expose it too; other dialog popup methods also have such argument.

Reply:
Looks like that argument is not exposed again (the third one).

Reply:
The search box will filter and select the specified type, so it effectively acts like currently selected type. You can test this by using Change Type in Scene dock, it will always select the type of the node you are replacing.